File: backend/TransportManagementSystem/Entity/Fuel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TransportManagementSystem.Entity
{
    public class Fuel
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [ForeignKey("Truck")]
        public int TruckId { get; set; }

        public Truck? Truck { get; set; }
        [Required]
        [ForeignKey("Driver")]
        public int DriverId { get; set; }

        public Driver? Driver { get; set; }

        [Required]
        public DateTime? FillDate { get; set; }
        
        public int? Quantity { get; set; }
    
        public string? OdometerReading { get; set; }
 
        public float? Amount { get; set; }
        public string? Comment { get; set; }

        public string? FuelTank { get; set; }

        [ForeignKey("FuelVendor")]
        public int FuelVendorId { get; set; }
        public FuelVendor? FuelVendor { get; set; }
    }
}

File: frontend/transport-management-system-web/src/app/types/fuel.ts
export interface IFuel {
  id: number;
  truckId: number;
  driverId: number;
  fillDate: string;
  quantity: number;
  odometerReading: string;
  amount: number;
  comment: string;
  fuelTank: string;
  fuelVendorId: number;
  
 
  truck?: {
    id: number;
    immatriculation: string;
    brand: string;
  };
  
  driver?: {
    id: number;
    name: string;
    permisNumber: string;
  };
  
  fuelVendor?: {
    id: number;
    name: string;
  };
}